src/components/button/button.spec.tsx

```diff
@@ -1,17 +1,150 @@
-import { render } from '@testing-library/react';
+import 'jest-styled-components';
+import { render, fireEvent, screen, findByText, getByRole } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 import { AuthProvider } from '../../helpers';
-import { lightTheme } from '../../styles';
+import { lightTheme, darkTheme } from '../../styles';
 import { Button } from './button';
+import { pxToRem } from '../../styles';
 
 describe('Button Component', () => {
 	it('should render a default button', () => {
+
+		const { getByText, getByRole } = render(
+			<AuthProvider>
+				<Button onClick={() => console.log('test')}>Primary default Button</Button>
+			</AuthProvider>
+		);
+		expect(getByText(/Primary default Button/)).toHaveStyle(
+			`background-color: ${lightTheme.button.default}; color: #FFF; border: 1px solid transparent; max-width: ${pxToRem(428)}; cursor: pointer;`
+		);
+
+		expect(getByText(/Primary default Button/)).toMatchSnapshot();
+
+		const btn = getByText(/Primary default Button/)
+		userEvent.hover(btn);
+		expect(btn).toHaveStyleRule("opacity", "0.8", {
+			modifier: ':hover',
+		});
+	});
+
+	it('should render a secondary default button', () => {
+		const { getByText } = render(
+			<AuthProvider>
+				<Button onClick={() => console.log('test')} variant={'secondary'}>Secondary default Button</Button>
+			</AuthProvider>
+		);
+		expect(getByText(/Secondary default Button/)).toHaveStyle(
+			`background-color: transparent; color: ${lightTheme.button.default}; border: 1px solid ${lightTheme.button.default}; max-width: ${pxToRem(160)}`
```

Comment:
What we test in `toHaveStyle` is repetetive => can you try to create a function / helper / test for that so we don't repeat ourselves all the time?